### PR TITLE
For python 3 and python2 compatibility

### PR DESCRIPTION
**Pull request by 3togo**
Thank you for your good work.

I did some minor modification to make your codes compatible to both python2 and python3

Have a nice day

Joe

### CODE REVIEW

**Comment by 3togo** [comment]
I use 2to3.py to translate. I did check and find that it is functioning properly. print(()) is due to the 2to3 translation. 18.08.2018, 17:24, "joha2" <notifications@github.com>:First of all thanks for your great work! Before accepting your PR, I've got a few questions:

Did you translate the code via script?
How do you verify that the translated code will do the same afterwards?
Why are the print() functions substituted by print(())?

We will check your code, which may take a while and afterwards accepting your PR.
Best wishes
Johannes

**Comment by joha2** [comment]
Thanks for cleaning up our messy code :-) I also thought that you did it via script, just wanted to be sure. I find these relative imports quite ugly, do you have any suggestion how to avoid them in a proper pythonic manner?

**Comment by 3togo** [comment]
To make the code compatible to both python 2 and 3,  there are two options as far as I know. 
Let's  use raster.py in sampling2d directory as an example.
you could either write full relative path:
from pyrateoptics.sampling2d import pds 
or
from . import pds

If there were one day that you want to change your directory name, the latter will be a better choice because you don't need to retype anything.  

Joe

**Comment by joha2** [comment]
Ah thanks for the explanation. I realized that the from . import pds not works if you have code in the __main__ part of your library. This needs the full import path. Is this correct?

**Comment by 3togo** [comment]
"from . import xxx " means for this prg directory.  Without "from .",  there is no way to tell whether you want to import xxx from a location relative to prg directory or relative to site-package.      25.08.2018, 16:41, "joha2" <notifications@github.com>:Ah thanks for the explanation. I realized that the from . import pds not works if you have code in the main part of your library. This needs the full import path. Is this correct?

File: pyrateoptics/raytracer/helpers.py [hunk 1]
```diff
@@ -103,16 +103,14 @@ def build_pilotbundle(surfobj, mat, (dx, dy), (phix, phiy), Elock=None,
     param random_xy: (bool) choose xy distribution randomly?
     
     """
-
```
Comment (joha2) [inline pyrateoptics/raytracer/helpers.py]: These lines were presumably accidentially removed. Please reinsert them into the code. Thank you :-)

File: pyrateoptics/raytracer/helpers.py [hunk 2]
```diff
@@ -191,17 +189,18 @@ def generate_cone_xy_bilinear(
...
+    (dx, dy) = xxx_todo_changeme5
+    (phix, phiy) = xxx_todo_changeme6
+    def generate_cone(direction_vec, lim_angle, xxx_todo_changeme2, num_pts_dir):
         """
         Generates cone for direction vector on real S_6 
         (x^2 + y^2 + z^2 + u^2 + v^2 + w^2 = 1)
         and generates cartesian raster for x and y
         """
-
-        # generate xy sampling and set pilot ray at position 0 in array
```
Comment (joha2) [inline pyrateoptics/raytracer/helpers.py]: This line was accidentially removed. Please reinsert it into the code. Thank you :-)

File: pyrateoptics/raytracer/helpers.py (hunk 1 above)
Comment (joha2) [reply pyrateoptics/raytracer/helpers.py]: (the TODOs in line 107 -- 109)

File: pyrateoptics/raytracer/helpers.py (hunk 2 above)
Comment (joha2) [reply pyrateoptics/raytracer/helpers.py]: (The comment in line 204)

**Pull request merged**